src/dune_engine/scheduler.ml

```diff
@@ -610,10 +650,13 @@ let with_job_slot f =
 
 let wait_for_process pid =
   let ivar = Fiber.Ivar.create () in
-  Process_watcher.register_job { pid; ivar };
+  let t = Fiber.Var.get_exn t_var in
+  Process_watcher.register_job t.process_watcher { pid; ivar };
   Fiber.Ivar.read ivar
 
-let wait_for_dune_cache () = Event.flush_dedup ()
+let wait_for_dune_cache () =
+  let t = Fiber.Var.get_exn t_var in
```

Comment:
There seem to be test failures on this line. Probably this function ends up being called without t_var being set?

Reply:
I fixed this by using the global var as we did earlier. `wait_for_dune_cache` is a bit weird. It seems like it would make more sense to have an ivar that is filled when no all the dedups are flushed and to have that run concurrently with the builds.